akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala

```diff
@@ -80,8 +81,60 @@ class GraphStageLogicSpec extends AkkaSpec with GraphInterpreterSpecKit with Con
     }
   }
 
+  final case class ReadNEmitN(n: Int) extends GraphStage[FlowShape[Int, Int]] {
```

Comment:
Well now you have a proper class, not an anonymous one. Makes things simpler to read because the case class has a proper toString method. But honestly, I don't know what @viktorklang s reason was, I just make things up :)